### PR TITLE
Move some severity configurations to globalconfig

### PR DESCRIPTION
**Pull request by Youssef1313**
See https://github.com/dotnet/roslyn/pull/65274#issuecomment-1315996877

### CODE REVIEW

**Comment by 333fred** [comment]
@Youssef1313 can you add a description? What is this PR supposed to accomplish?

**Comment by Youssef1313** [comment]
> @Youssef1313 can you add a description? What is this PR supposed to accomplish?

Use of globalconfigs was something recommended by @sharwell as being more efficient (see https://github.com/dotnet/aspnetcore/pull/30138#discussion_r575769897), I don't know why it is more efficient though.

File: .editorconfig [hunk 1]
```diff
@@ -289,9 +262,6 @@ dotnet_diagnostic.IDE2001.severity = warning
 # csharp_style_allow_blank_lines_between_consecutive_braces_experimental
 dotnet_diagnostic.IDE2002.severity = warning
 
-# dotnet_style_allow_statement_immediately_after_block_experimental
-dotnet_diagnostic.IDE2003.severity = warning
```
Comment (Youssef1313) [inline .editorconfig:293]: This has no effect 😕 

The corresponding code-style option default value is true which means "disable the analyzer":

https://github.com/dotnet/roslyn/blob/85c8cf51143af5ac9b782ff959aa286b4a6e2e38/src/Analyzers/Core/Analyzers/NewLines/ConsecutiveStatementPlacement/AbstractConsecutiveStatementPlacementDiagnosticAnalyzer.cs#L41-L43

I guess it's not intentional, @CyrusNajmabadi ?

**Comment by 333fred** [comment]
> Use of globalconfigs was something recommended by @sharwell as being more efficient (see https://github.com/dotnet/aspnetcore/pull/30138#discussion_r575769897), I don't know why it is more efficient though.

We'll need that answered :).

**Comment by 333fred** [comment]
Marking as draft pending a more complete description of what this solves.

**Comment by Youssef1313** [comment]
@sharwell Can you advise here?

**Comment by sharwell** [comment]
**.globalconfig** only needs to be evaluated once for a project, while **.editorconfig** needs to be evaluated for each document. @mavasani can describe what optimizations are applied regarding enabling/disabling analyzers when the configuration is provided for the entire project.

**Comment by Youssef1313** [comment]
@333fred Should this be ready to merge?

**Review by sharwell — APPROVED** [review]

File: .editorconfig [hunk 2]
```diff
@@ -254,9 +224,6 @@ csharp_prefer_braces = true:silent
 csharp_preserve_single_line_blocks = true
 csharp_preserve_single_line_statements = true
 
-# IDE0060: Remove unused parameter
-dotnet_diagnostic.IDE0060.severity = warning
```
[the remark this answers is not in the document]
Comment (333fred) [reply .editorconfig:258]: @Youssef1313 I don't think we should make this change as part of this PR. Please move *.cs things back to just *.cs, and revert the VB changes.

**Review by 333fred — APPROVED** [review]

**Comment by Youssef1313** [comment]
@333fred Can we get this merged? Thanks!

**Pull request merged**